### ISSUE TITLE
Directory structure - Implement new `spacetimedb-update` cli

### ISSUE THREAD

**Issue by bfops** [clockworklabs/SpacetimeDB#1839]
# Background
We would like a new `spacetimedb-update` binary:
> The new spacetimedb-update binary will be used to manage versions of the other 3 binaries. Any commands not recognized by the spacetimedb-update binary will be forwarded on to the spacetimedb-cli version that is set as the current version of the CLI for the user. If a command is not recognized by spacetimedb-cli, it should look for an executable named spacetimedb-{subcommand} and execute that, passing in the arguments it received for the subcommand.

> Under this scheme, the spacetimedb-update binary will be aliased to spacetime and will be placed on the user's path so it can be executed as spacetime.

# Definition of "Done"
TODO

**Comment by gefjon** [comment]
@coolreader18 to add tests. @bfops to review.

**Comment by gefjon** [comment]
From @bfops out-of-band:

Testing
I don't have a strong opinion about whether we do extensive testing before or after the PR merges, since this PR mostly implements functionality that was behind unimplemented! before, so it's not really likely to be "more broken" if we merge the PR.

This is what I have jotted down in terms of testing. I think we should test all of these on Linux, MacOS, and Windows:
installing a new version
confirming that it works with hotfix versions
spacetimedb-update gets updated if it's a newer version
useing an installed version
unintalling a current version
The binaries-delegating-to-other-binaries flow
spacetimedb-cli version -> spacetimedb-update
Running spacetimedb-update directly
spacetimedb-update delegates to CLI if it's run as spacetime
spacetimedb-cli delegates to the appropriate server binary
Listing installed versions
Inferring which version is current

Yeah I'm curious about how we do the testing, since I think cargo run and cargo install don't work the same as a "production" install flow... but the current github releases don't work with the new spacetimedb-update install (name changes), so I'm not sure how to approach getting this tested in a production-like way.

I did test some things on my machine like listing the versions, inferring the current version, etc., but I was setting up the directory structures manually so it wasn't really a representative test.

**Comment by gefjon** [comment]
@bfops suggests mocking out the GitHub part, so that it fetches from localhost or something instead in testing.

**Comment by bfops** [comment]
Yeah generally I feel like automated tests can test well something of the form "_given_ that the external part works in xyz way, the rest of the system works as expected". Then the "external part works in xyz" is more reasonable to test manually (for example here, the upgrade flow would only break if the GitHub API itself changes)

**Comment by bfops** [comment]
I'm going to reopen this to track the "test and make sure it works" part of things.
